tests/models/decoder_only/vision_language/vlm_utils/case_filtering.py

```diff
@@ -0,0 +1,154 @@
+            # For all types besides custom inputs, we need a prompt formatter
+            else:
+                assert test_info.prompt_formatter is not None
+
+            # Everything looks okay; keep if this is has correct proc handling
+            if (test_info.distributed_executor_backend
+                    is not None) == fork_per_test:
+                matching_tests[test_name] = test_info
+                # Check if it's skipped or not
+                if test_info.skip:
+                    skipped_names.append(test_name)
+
+    return matching_tests, skipped_names
+
+
+def get_parametrized_options(test_settings: Dict[str, VLMTestInfo],
+                             test_type: VLMTestType,
+                             fork_new_process_for_each_test: bool):
+    """Converts all of our VLMTestInfo into an expanded list of parameters.
+    This is similar to nesting pytest parametrize calls, but done directly
+    through an itertools product so that each test can set things like
+    size factors etc, while still running in isolated test cases.
+    """
+    matching_tests, skipped_names = get_filtered_test_settings(
+        test_settings, test_type, fork_new_process_for_each_test)
+
+    # Ensure that something is wrapped as an iterable it's not already
+    ensure_wrapped = lambda e: e if isinstance(e, (list, tuple)) else (e, )
+
+    def get_model_type_cases(model_type: str, test_info: VLMTestInfo):
+        # This is essentially the same as nesting a bunch of mark.parametrize
+        # decorators, but we do it programmatically to allow overrides for on
+        # a per-model basis, while still being able to execute each of these
+        # as individual test cases in pytest.
+        iterables = [
+            ensure_wrapped(model_type),
+            ensure_wrapped(test_info.models),
+            ensure_wrapped(test_info.max_tokens),
+            ensure_wrapped(test_info.num_logprobs),
+            ensure_wrapped(test_info.dtype),
+            ensure_wrapped(test_info.distributed_executor_backend),
+        ]
+        # num_frames is video only
+        if test_type == VLMTestType.VIDEO:
+            iterables.append(ensure_wrapped(test_info.num_video_frames))
+
+        # No sizes passed for custom inputs, since inputs are directly provided
+        if test_type != VLMTestType.CUSTOM_INPUTS:
+            iterables.append(get_wrapped_test_sizes(test_info, test_type))
+        #Otherwise expand the custom test options instead
+        else:
+            if test_info.custom_test_opts is None:
+                raise ValueError("Test has type CUSTOM_INPUTS, but none given")
+
+            iterables.append(test_info.custom_test_opts)
+
+        # Wrap all model cases in a pytest parameter & explicitly skip anything
+        # that had a met skip condition, but otherwise matched the filter.
+        return [
+            pytest.param(
+                *case,
+                marks=pytest.mark.skipif(
```

Comment:
Can you extend this so we can provide custom marks? For example, we may need to mark particular tests as xfail or tag certain tests so only a subset are run in CI.

Reply:
This should also simplify the logic as you don't have to manage `skipped_names` anymore (we can provide a `pytest.mark.skipif` directly into `VLMTestInfo`.